CMakeLists.txt

```diff
@@ -408,18 +408,27 @@ if(UNIX)
   include(CheckSymbolExists)
 
   # don't allow implicit function declarations
-  set(CMAKE_REQUIRED_FLAGS "-std=c99")
-  set(CMAKE_REQUIRED_LIBRARIES "rt")
+  set(CMAKE_REQUIRED_FLAGS "-std=c99 -Werror=implicit-function-declaration")
+  if (NOT APPLE)
+    set(CMAKE_REQUIRED_LIBRARIES "rt")
+  endif ()
 
   set(CMAKE_REQUIRED_DEFINITIONS "-D_POSIX_C_SOURCE=200112L")
   CHECK_SYMBOL_EXISTS("posix_memalign"
                       "stdlib.h"
                       HAVE_POSIX_MEMALIGN)
 
   set(CMAKE_REQUIRED_DEFINITIONS "-D_POSIX_C_SOURCE=199309L")
-  CHECK_SYMBOL_EXISTS("clock_gettime"
-                      "time.h"
-                      HAVE_CLOCK_GETTIME)
+
+  CHECK_C_SOURCE_COMPILES("
+#include <time.h>
+
+int main(int argc, char** argv) {
+  struct timespec timeout = {0, 0};
+  clock_gettime(CLOCK_REALTIME, &timeout);
```

Comment:
Should this have `futimens(fd, NULL)`? There's code like
```
#ifdef HAVE_CLOCK_GETTIME
    futimens(fd, NULL);
#else
#endif
```
in LLVMFileUtils.cc

In the build that I said was successful I'm doing cmake `-DHAVE_CLOCK_GETTIME=0`

Reply:
Added a new check `HAVE_FUTIMENS`